### PR TITLE
Catch and log ValueError exceptions

### PR DESCRIPTION
**Pull request by darindf**
Found that there are ValueError exceptions when validating results using 3 different classifiers. Added try/exception block to catch these errors, and to log these errors.

Modified logger class to have support an exception method, which is standard when using default python loggers.

### CODE REVIEW

**Comment by omri374** [comment]
Hi @darindf, Thanks for your contribution!

1. Please add unit tests for the updated code
2. Please make sure the pytest, flake8 and pylint complete successfully (see the contribution guidelines here: https://github.com/microsoft/presidio/blob/master/CONTRIBUTING.md#how-to-test)
Otherwise the build won't succeed and we will not be able to approve the PR.

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/credit_card_recognizer.py [hunk 1]
```diff
@@ -31,22 +31,29 @@ def __init__(self):
...
         odd_digits = digits[-1::-2]
         even_digits = digits[-2::-2]
         checksum = 0
         checksum += sum(odd_digits)
         for d in even_digits:
-            checksum += sum(digits_of(d * 2))
+            try:
+                checksum += sum(digits_of(d * 2))
+            except ValueError:
+                self.logger.exception(f'int conversion failed for sanitized_value: {sanitized_value} digits: {digits} even_digits: {even_digits}')
```
Comment (eladiw) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/credit_card_recognizer.py]: Not sure this is how it done all over, however, logging possible PII is not recommended.
I'd suggest to remove the values from thhe error msg

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/iban_recognizer.py [hunk 2]
```diff
@@ -50,11 +50,15 @@ def validate_result(self, pattern_text):
...
-    def __generate_iban_check_digits(iban):
+    # @staticmethod
+    def __generate_iban_check_digits(self,iban):
         transformed_iban = (iban[:2] + '00' + iban[4:]).upper()
         number_iban = IbanRecognizer.__number_iban(transformed_iban)
-        return '{:0>2}'.format(98 - (int(number_iban) % 97))
+        try: 
+            return '{:0>2}'.format(98 - (int(number_iban) % 97))
+        except ValueError:
+            self.logger.exception(f'int conversion failed for iban: {iban} transformed_iban: {transformed_iban} number_iban: {number_iban}')
```
Comment (eladiw) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/iban_recognizer.py]: same comment regarding the PII logging

File: presidio-analyzer/presidio_analyzer/predefined_recognizers/uk_nhs_recognizer.py [hunk 3]
```diff
@@ -26,6 +26,7 @@ def validate_result(self, pattern_text):
             try:
                 val = int(c)
             except ValueError:
+                self.logger.exception(f'int conversion failed for pattern_text: {pattern_text} text: {text}')
```
Comment (eladiw) [inline presidio-analyzer/presidio_analyzer/predefined_recognizers/uk_nhs_recognizer.py]: same comment regarding PII logging

**Pull request closed without merging**